measurements/api/measurements.py

```diff
@@ -323,67 +332,71 @@ def list_measurements(
     iter_start_time = time.time()
     results = []
 
-    try:
-        for row in q:
-            measurement_id = '{}-{}'.format(MSM_ID_PREFIX, row.msm_no)
-            url = urljoin(
+    with configure_scope() as scope:
+        scope.set_extra("sql_query", str(q))
+
+        try:
+            for row in q:
+                measurement_id = '{}-{}'.format(MSM_ID_PREFIX, row.msm_no)
+                url = urljoin(
+                    current_app.config['BASE_URL'],
+                    '/api/v1/measurement/%s' % measurement_id
+                )
+                results.append({
+                    'measurement_url': url,
+                    'measurement_id': measurement_id,
+                    'report_id': row.report_id,
+                    'probe_cc': row.probe_cc,
+                    'probe_asn': "AS{}".format(row.probe_asn),
+                    'test_name': row.test_name,
+                    'measurement_start_time': row.measurement_start_time,
+                    'input': row.input,
+                    'anomaly': row.anomaly,
+                    'confirmed': row.confirmed,
+                    'failure': (row.exc != None
+                                or row.residual_no != None
+                                or row.msm_failure),
+                })
+        except exc.OperationalError as exc:
+            if isinstance(exc.orig, QueryCanceledError):
+                raise QueryTimeoutError()
```

Comment:
I changed the error capturing to be a bit more precise (we actually should trap the sqlalchemy sub exception) and made a custom exception to make filtering in the sentry UI easier.